### PR TITLE
Add hostNetwork DNSPolicy support

### PR DESCRIPTION
**Pull request by zzjin**
And add support listen port under 1024 (like 443).

**What type of PR is this?**
fix: fix new introduced `hostNetwork` cannot speak to xds bug.

May fix: #2310

### CODE REVIEW

**Review by Xunzhuo — CHANGES_REQUESTED** [review]
Plz fix the CI and make sure it is backport compatible.

File: internal/gatewayapi/listener.go [hunk 1]
```diff
@@ -96,7 +96,15 @@ func (t *Translator) ProcessListeners(gateways []*GatewayContext, xdsIR XdsIRMap
 
 			// Add the listener to the Xds IR
 			servicePort := &protocolPort{protocol: listener.Protocol, port: int32(listener.Port)}
-			containerPort := servicePortToContainerPort(servicePort.port)
+
+			containerPort := servicePort.port
```
Comment (shawnh2) [inline internal/gatewayapi/listener.go:100]: just out of curiosity, is there any better way to improve this looong hideous nested nil check?

File: internal/gatewayapi/listener.go [hunk 2]
```diff
@@ -96,7 +96,15 @@ func (t *Translator) ProcessListeners(gateways []*GatewayContext, xdsIR XdsIRMap
...
+			if !(resources.EnvoyProxy != nil && resources.EnvoyProxy.Spec.Provider != nil &&
```
Comment (shawnh2) [inline internal/gatewayapi/listener.go:101]: just out of curiosity, is there any better way to improve this looong hideous nested nil check?

**Comment by shawnh2** [comment]
hi @zzjin, you can try `make testdata` to automatically generate the correct test data.

File: internal/gatewayapi/listener.go (hunk 2 above)
Comment (Xunzhuo) [reply internal/gatewayapi/listener.go:101]: refer to https://github.com/envoyproxy/gateway/blob/main/api/v1alpha1/envoygateway_helpers.go#L80

make it as a common helper to use

**Comment by arkodg** [comment]
we shouldnt be changing the container port logic, all clients should access the gateway via the service

**Comment by Xunzhuo** [comment]
@arkodg Any better solutions for using EG in such scenarios:

1. a kubernetes cluster without lb implementation
2. expose envoyproxy with 80/443

In this scenario, EG is more like a real loadbalancer, it should be a common request.

**Comment by arkodg** [comment]
@Xunzhuo you can route to the service port (80/443) via External IP

**Comment by Xunzhuo** [comment]
@arkodg Can you elaborate it ?

**Comment by arkodg** [comment]
> @arkodg Can you elaborate it ?

https://www.fadhil-blog.dev/blog/kubernetes-external-ip/

**Comment by Xunzhuo** [comment]
Correct ... @arkodg I think I missed this, thanks for pointing out.

**Comment by Xunzhuo** [comment]
close via https://github.com/envoyproxy/gateway/pull/2374

**Pull request closed without merging**